### ISSUE TITLE
[Develop] Body Coupling

### ISSUE THREAD

**Issue by geggleto** [slimphp/Slim#1269]
https://github.com/slimphp/Slim/blob/develop/Slim/App.php#L314

``` php
            if ($hasBody) {
                $body = $response->getBody(); //Returns the PSR Steam Interface
                if ($body->isAttached()) { //This method is NOT a part of the interface but is apart of the Slim\Http\Body
                    $body->rewind();
                    while (!$body->eof()) {
                        $settings = $this->container->get('settings');
                        echo $body->read($settings['responseChunkSize']);
                    }
                }
            }
```

**Comment by codeguy** [comment]
Good catch. I want to say that existed at one time, else it's a remnant from an earlier Guzzle stream implementation. Either way, this needs to go.

**Comment by geggleto** [comment]
@codeguy Is this part of the code responsible for outputting the response body ?

**Comment by codeguy** [comment]
Yes, it is the last stage in the app lifecycle which returns the content out of PHP to the upstream web server and on to the HTTP client. All `isAttached()` does is make sure there is still an underlying resource attached to the StreamInterface instance.

**Comment by geggleto** [comment]
I am going to wait to action this as I know the Slim\Http\Body object is changing with the UploadedFiles PSR implementation.

**Comment by codeguy** [comment]
@geggleto Sounds good

**Comment by akrabat** [comment]
Fixed via #1317
